app/code/Bulbulatory/Recomendations/Model/Total/Quote/Custom.php

```diff
@@ -0,0 +1,53 @@
+<?php
+namespace Bulbulatory\Recomendations\Model\Total\Quote;
+
+use Magento\Quote\Model\Quote\Address\Total\AbstractTotal;
+use Magento\Framework\Pricing\PriceCurrencyInterface;
+use Magento\Quote\Model\Quote;
+use Magento\Quote\Api\Data\ShippingAssignmentInterface;
+use Magento\Quote\Model\Quote\Address\Total;
+use Bulbulatory\Recomendations\Block\Recommendation\Index;
+
+class Custom extends AbstractTotal
+{
+   protected $_priceCurrency;
+   protected $index;
+
+   public function __construct(
+      PriceCurrencyInterface $priceCurrency,
+      Index $index
+   ){
+       $this->_priceCurrency = $priceCurrency;
+       $this->index = $index;
+   }
+
+   public function collect(
+       Quote $quote,
+       ShippingAssignmentInterface $shippingAssignment,
+       Total $total
+   )
+   {
+        parent::collect($quote, $shippingAssignment, $total);
+
+        $percent = $this->index->calculateDiscountPercent();
```

Comment:
Można tutaj sprawdzić, czy jakakolwiek zniżka została zwrócona - jeśli nie, to nie ma sensu iść dalej i ustawiać wartości, skoro są one zerowe i nie wpływają na obecną wartość koszyka.